apk2java

```diff
@@ -10,70 +11,72 @@ echo "This script will work on automating the work of extracting the source code
 if [[ $# -eq 0 || $1 == "--help" || $1 == "-help" ]]
 	then
 	echo ""
-	echo "$0 APK_FILE_NAME"
+	echo "$0 APK_FILE_NAME DESTINATION"
 	exit 
 fi
-echo "Starting APK Decompile"
+echo "[*] Starting APK Decompile"
 # JAR KEEP is a variable which allows you to keep jar file converted using dex2jar.
 # 0 is off
 # 1 is on
 JAR_KEEP=1
-echo $0 $1
-echo "$@"
-echo "APK to JAVA/SRC conversion Utility"
-APK_NAME=`basename $1`
-c=`basename $APK_NAME .apk`
-FULL_PTH=`readlink -f $1`
+APK_NAME=`basename "$1"`
+c=`basename "$APK_NAME" .apk`
+FULL_PTH=`readlink -f "$1"`
 CDIR=`dirname "$FULL_PTH"`
-SRC_DIR=$CDIR"/"$APK_NAME"_src/"
-SRC_PATH=$CDIR"/"$APK_NAME"_src/"
-echo $c
-echo $APK_NAME
-echo "FULL_PATH"$FULL_PTH
-echo "CDIR"$CDIR
-if [ $c == `basename $APK_NAME` ]
+
+if [[ $# -eq 2 ]]
+then
+	if [[ "$2" != /* ]]
```

Comment:
this is where we need to see if the $2 is relative or full path and then append the full path. may be realpath or grealpath (in case of mac) in this case would be helpful.